### PR TITLE
Add web routes for ingress service

### PR DESCRIPTION
**Pull request by inickles-grapl**
### What changes does this PR make to Grapl? Why?

This creates a web API to expose the ingress service for publishing logs to Grapl.

This includes an integration test and enables the E2E tests to use a public API instead of hitting this Ingress Service directly.

### How were these changes tested?

CI

### CODE REVIEW

File: nomad/grapl-core.nomad [hunk 1]
```diff
@@ -627,7 +627,8 @@ job "grapl-core" {
         GRAPL_USER_AUTH_TABLE    = var.user_auth_table
         GRAPL_USER_SESSION_TABLE = var.user_session_table
 
-        PLUGIN_REGISTRY_CLIENT_ADDRESS = "http://${NOMAD_UPSTREAM_ADDR_plugin-registry}"
+        PLUGIN_REGISTRY_CLIENT_ADDRESS  = "http://${NOMAD_UPSTREAM_ADDR_plugin-registry}"
+        PIPELINE_INGRESS_CLIENT_ADDRESS = "http://${NOMAD_UPSTREAM_ADDR_pipeline-ingress}"
```
Comment (wimax-grapl) [inline nomad/grapl-core.nomad:631]: You need to add this as an upstream to your `service {}`

**Review by wimax-grapl — APPROVED** [review]

**Review by wimax-grapl — CHANGES_REQUESTED** [review]
whoops didnt mean to approve

File: src/rust/grapl-web-ui/tests/api/ingress.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    app.login_with_test_user().await?;
+
+    let plugin_name = "Test Plugin";
+
+    let create_response = crate::plugin::create_plugin(&app, plugin_name).await?;
+
+    //TODO: this shouldn't be necessary, but we're seeing 50 errors without it.
+    // I'll file a task to look into it for now so we can unblock frontend work.
+    std::thread::sleep(std::time::Duration::from_secs(5));
```
Comment (wimax-grapl) [inline src/rust/grapl-web-ui/tests/api/ingress.rs:19]: like fifty separate errors or 50x errors?

Comment (wimax-grapl) [reply src/rust/grapl-web-ui/tests/api/ingress.rs:19]: dont let this comment prevent you from shipping though its an easy fix later if it's 50x

**Review by wimax-grapl — APPROVED** [review]

File: src/rust/grapl-web-ui/tests/api/ingress.rs (hunk 2 above)
Comment (inickles-grapl) [reply src/rust/grapl-web-ui/tests/api/ingress.rs]: It bothers me, I can wait for another CI test run. 👍

**Pull request merged**